lib/project_types/extension/features/argo_config.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Extension
+  module Features
+    class ArgoConfig
+      CONFIG_FILE_NAME = 'extension.config.yml'
+
+      def self.parse_yaml(context)
+        require 'yaml' # takes 20ms, so deferred as late as possible.
```

Comment:
Would it make sense to check for the existence of this file first and return `{}` if it does not exist?  This would handle the missing file case and skip loading the `yaml` if there is no file to parse.

Reply:
Good idea, changed it.